knode/validator/validator.go

```diff
@@ -642,9 +649,15 @@ func (val *validator) updateValidators(checksum [32]byte, genesis bool) error {
 }
 
 func (val *validator) Deposits() ([]*types.Deposit, error) {
+	if !val.Validating() {
+		return nil, ErrIsNotRunning
+	}
 	return val.consensus.Deposits(val.walletAccount.Account().Address)
 }
 
 func (val *validator) RedeemDeposits() error {
+	if !val.Validating() {
```

Comment:
same here

Reply:
It uses `val.walletAccount` too in more details, this is described below.